oci/archive/oci_transport_test.go

```diff
@@ -107,11 +127,56 @@ func TestNewReference(t *testing.T) {
 
 	_, err = NewReference(tmpDir+"/has:colon", imageValue)
 	assert.Error(t, err)
+
+	// Test private newReference
+	_, err = newReference(tmpDir, "imageName", 1, nil, nil) // Both image and sourceIndex specified
+	assert.Error(t, err)
+}
+
+func TestNewIndexReference(t *testing.T) {
+	const imageValue = "imageValue"
+
+	tmpDir, err := ioutil.TempDir("", "oci-transport-test")
+	require.NoError(t, err)
+	defer os.RemoveAll(tmpDir)
+
+	ref, err := NewIndexReference(tmpDir, 10)
+	require.NoError(t, err)
+	ociArchRef, ok := ref.(ociArchiveReference)
+	require.True(t, ok)
+	assert.Equal(t, tmpDir, ociArchRef.file)
+	assert.Equal(t, "", ociArchRef.image)
+	assert.Equal(t, 10, ociArchRef.sourceIndex)
+
+	ref, err = NewIndexReference(tmpDir, 9999)
+	require.NoError(t, err)
+	ociArchRef, ok = ref.(ociArchiveReference)
+	require.True(t, ok)
+	assert.Equal(t, tmpDir, ociArchRef.file)
+	assert.Equal(t, "", ociArchRef.image)
+	assert.Equal(t, 9999, ociArchRef.sourceIndex)
+
+	_, err = NewIndexReference(tmpDir+"/thisparentdoesnotexist/something", 10)
+	assert.Error(t, err)
+
+	// sourceIndex cannot be less than -1
+	_, err = NewIndexReference(tmpDir, -3)
+	assert.Error(t, err)
+
+	_, err = NewIndexReference(tmpDir+"/has:colon", 99)
+	assert.Error(t, err)
+
+	// Test private newReference
+	_, err = newReference(tmpDir, imageValue, 1, nil, nil)
+	assert.Error(t, err)
 }
 
 // refToTempOCI creates a temporary directory and returns an reference to it.
-func refToTempOCI(t *testing.T) (types.ImageReference, string) {
-	tmpDir := t.TempDir()
+// The caller should
+//   defer os.RemoveAll(tmpDir)
+func refToTempOCI(t *testing.T, sourceIndex bool) (ref types.ImageReference, tmpDir string) {
+	tmpDir, err := ioutil.TempDir("", "oci-transport-test")
+	require.NoError(t, err)
```

Comment:
(Merge problem) The use of `t.TempDir` makes the caller cleaning up unnecessary.